wal/src/message_queue_impl/region.rs

```diff
@@ -163,29 +181,63 @@ pub struct Region<M: MessageQueue> {
 impl<M: MessageQueue> Region<M> {
     /// Init the region.
     pub async fn open(namespace: &str, region_id: RegionId, message_queue: Arc<M>) -> Result<Self> {
+        info!(
+            "Begin to open region in namespace, namespace:{}, region id:{}",
+            namespace, region_id
+        );
+
+        let log_encoding = CommonLogEncoding::newest();
+        let meta_encoding = MetaEncoding::newest();
+
         // Format to the topic name.
         let log_topic = format_wal_data_topic_name(namespace, region_id);
         let meta_topic = format_wal_meta_topic_name(namespace, region_id);
         message_queue
             .create_topic_if_not_exist(&log_topic)
             .await
             .map_err(|e| Box::new(e) as _)
-            .context(Open {
+            .context(OpenWithCause {
                 namespace,
                 region_id,
-            });
+                msg: "failed while trying to create topic",
+            })?;
         message_queue
             .create_topic_if_not_exist(&meta_topic)
             .await
             .map_err(|e| Box::new(e) as _)
-            .context(Open {
+            .context(OpenWithCause {
                 namespace,
                 region_id,
-            });
+                msg: "failed while trying to create topic",
+            })?;
+
+        // Build region meta.
+        let mut region_meta_builder = RegionMetaBuilder::default();
+        let high_watermark_in_snapshot = Self::recover_region_meta_from_meta(
+            namespace,
+            region_id,
+            &mut region_meta_builder,
+            message_queue.as_ref(),
+            &meta_topic,
+            &meta_encoding,
```

Comment:
Seems no need to pass the `meta_encoding` to the method, and why not just declare it in the method?

Reply:
`SnapshotSynchronizer`'s initialization need `MetaEncoding`, too.
So I think define here and pass is better.